nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/io/LogEntryCSVWriterTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.platform.audit.io;
+
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.PathRef;
+import org.nuxeo.ecm.core.io.marshallers.csv.AbstractCSVWriterTest;
+import org.nuxeo.ecm.core.io.marshallers.csv.CSVAssert;
+import org.nuxeo.ecm.core.io.registry.context.RenderingContext;
+import org.nuxeo.ecm.platform.audit.AuditFeature;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.query.api.PageProvider;
+import org.nuxeo.ecm.platform.query.api.PageProviderService;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+
+/**
+ * @since 11.1
+ */
+@Features(AuditFeature.class)
+@Deploy("org.nuxeo.ecm.platform.audit.tests:test-pageprovider-contrib.xml")
+public class LogEntryCSVWriterTest extends AbstractCSVWriterTest.External<LogEntryCSVWriter, LogEntry> {
+
+    @Inject
+    private PageProviderService pps;
+
+    @Inject
+    private CoreSession session;
+
+    public LogEntryCSVWriterTest() {
+        super(LogEntryCSVWriter.class, LogEntry.class);
+    }
+
+    @Test
+    public void testDefaultProperties() throws IOException {
+        DocumentModel root = session.getDocument(new PathRef("/"));
```

Comment:
Can you create a real document to export and assert the csv field values?

Reply:
I do check all fields on a LogEntry from a file now.
CSVAssert doesn't give enough apis to check on everything so I just check the presence of the id and dates.